simon/tokio console support for core and repl tool (#6018)
deltachat/deltachat-core-rust: deltachat-repl/README.md

# Delta Chat REPL

> This is a simple [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop) frontend build on top of delta chat core.
> It's purpose is to help with quick testing during development, it is not meant for end users.

Reviewer: ```suggestion
This is a simple [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop) frontend built on top of Delta Chat core.
Its purpose is to help with quick testing during development, it is not meant for end users.
```

Dependencies:

> - if you want to use `getqr` you need `qrencode` (To install use your systems package manager)

Reviewer: ```suggestion
- if you want to use `getqr`, you need `qrencode` (To install, use your system's package manager)
```
Though i'm not good at punctuation, so don't trust me too much

Author: ```suggestion
- If you want to use `getqr` you need `qrencode` (To install, use your system's package manager)
```
language tool suggests this.

> - if you want to use `getqr` you need `qrencode` (macOS: `brew install qrencode`)

Reviewer: In Debian the package is also `qrencode`, and i think, everywhere, so per-OS instructions may be omitted.

Author: ```suggestion
- if you want to use `getqr` you need `qrencode` (To install use your systems package manager)
```

## Usage

```
cargo run <path to deltachat db>
```

Type in `help` to learn about what comands are available.

## Usage with `tokio-console`

> Tokio is the async runtime that delta chat core uses.
> Core uses tokio tasks, which is something similar to a thread.
> `tokio-console` is like a task manager for these tokio-tasks.

Reviewer: ```suggestion
Tokio is an async runtime that Delta Chat core uses.
Core uses Tokio tasks, which are something similar to threads.
`tokio-console` is like a task manager for these Tokio tasks.
```

Examples of tasks:

> - The event loop in the repl tool which processes events received from core

Reviewer: ```suggestion
- The event loop in the REPL tool which processes events received from core
```

> - The repl loop itself which waits for and executes user commands

Reviewer: ```suggestion
- The REPL loop itself which waits for and executes user commands
```

> - The imap task that manages imap connection in core

Reviewer: ```suggestion
- The IMAP task that manages IMAP connection in core
```

```
RUSTFLAGS="--cfg tokio_unstable" cargo run <path to deltachat db>
```

Then in a new console window start [`tokio-console`](https://github.com/tokio-rs/console).

> You can install it via `cargo install tokio-console`.

Author: ```suggestion
You can install it via `cargo install --locked tokio-console`.
```

### Quick Example

> An example session in the repl tool.

Reviewer: ```suggestion
An example session in the REPL tool:
```

```
RUSTFLAGS="--cfg tokio_unstable" cargo run test-db/db
setqr dcaccount:https://nine.testrun.org/new
configure
connect
listchats
getqr
```

> If it crashes you can just start it again and use the openpgp4fpr url instead of scanning the code from the terminal.

Reviewer: ```suggestion
If it crashes you can just start it again and use the openpgp4fpr URL instead of scanning the code from the terminal.
```

> 
> If it crashes you can just start it again and use the openpgp4fpr URL instead of scanning the code from the terminal.
> Or install `qrencode` to fix the crash and run `getqr` again.

Author: ```suggestion
```
for when https://github.com/deltachat/deltachat-core-rust/pull/6051 is merged

Reviewer 2: https://github.com/deltachat/deltachat-core-rust/pull/6051 is merged now

> Use the qrcode/openpgp4fpr link to setup the contact on deltachat.

Reviewer: ```suggestion
Use the qrcode/openpgp4fpr link to setup the contact on Delta Chat.
```

> Then write a message to that new contact, after that we can accept the chat in the repl tool and send a reply:

Reviewer: ```suggestion
Then write a message to that new contact, after that you can accept the chat in the REPL tool and send a reply:
```
